stores/colorCodePuzzleStore.js

```diff
@@ -0,0 +1,48 @@
+import { create } from "zustand";
+import { colors, getRandomColor, randomNumbers, symbols } from "../utils/utils";
+
+const useColorCodePuzzleStore = create((set) => ({
+  randomColors: [
+    colors[getRandomColor()],
+    colors[getRandomColor()],
+    colors[getRandomColor()],
+  ],
+  randomSymbols: [
+    symbols[randomNumbers[0]],
+    symbols[randomNumbers[1]],
+    symbols[randomNumbers[2]],
+  ],
+  colorCounts: { first: 2, second: 2, third: 2 },
+
+  resetRandomColors: () =>
+    set({
+      randomColors: [
+        colors[getRandomColor()],
+        colors[getRandomColor()],
+        colors[getRandomColor()],
+      ],
+    }),
+  resetRandomSymbols: () =>
+    set({
+      randomSymbols: [
+        symbols[[randomNumbers[0]]],
+        symbols[[randomNumbers[1]]],
+        symbols[[randomNumbers[2]]],
+      ],
+    }),
+  increaseFirstColorDiv: () =>
+    set((state) => ({
+      colorCounts: { first: state.colorCounts.first++, ...state.colorCounts },
+    })),
+  increaseSecondColorDiv: () =>
+    set((state) => ({
+      colorCounts: { second: state.colorCounts.second++, ...state.colorCounts },
+    })),
+  increaseThirdColorDiv: () =>
+    set((state) => ({
+      colorCounts: { third: state.colorCounts.third++, ...state.colorCounts },
+    })),
```

Comment:
These three functions are redundant and can be replaced with:

```js
increaseColorDiv: (color) =>
  set((state) => ({
    colorCounts: {
      ...state.colorCounts,
      [color]: state.colorCounts[color] + 1,
    },
  })),
```

You can call this function with:

```js
increaseColorDiv("first");
increaseColorDiv("second");
increaseColorDiv("third");
```